### PR TITLE
Drops should use content setters if they exist

### PR DESCRIPTION
**Pull request by parkr**
@evgenyneu, this fixes your issue as defined here: https://github.com/jekyll/jekyll/issues/4246#issuecomment-168367510

### CODE REVIEW

**Comment by evgenyneu** [comment]
Thank you, I have added this change, but it does not fix [my issue](https://github.com/jekyll/jekyll/issues/4246#issuecomment-168367510).

File: features/layout_data.feature [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  In order to make the layouts slightly dynamic
+
+  Scenario: Use custom layout data
+    Given I have a _layouts directory
+    And I have a "_layouts/custom.html" file with content:
+      """
+      ---
+      foo: my custom data
+      ---
+      {{ content }} foo: {{ layout.foo }}
```
Comment (evgenyneu) [inline features/layout_data.feature]: In my case I was doing `{{ layout.foo }}` from parent layout file `index.html`, not from `custom.html`.

Comment (parkr) [inline features/layout_data.feature]: See a9b80b2 for this. It works on master according to my scenario.

**Comment by evgenyneu** [comment]
Yes, you are right, @parkr, thank you. I needed to install jekyll from master to make it work.

**Pull request merged**